### PR TITLE
Migrate to GitHub Actions.

### PR DESCRIPTION
**Pull request by jezdez**
Travis CI has a new pricing model which places limits on open source.

* https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
* https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

Many projects are moving to GitHub Actions instead, including Jazzband projects: 

* https://github.com/orgs/jazzband/projects/1
* https://github.com/jazzband-roadies/help/issues/206

This is based on https://github.com/jazzband/contextlib2/pull/26.

It drops support for Python 3.5 because GitHub Actions doesn't fully support it and it's EOL anyway.

It also adds Python 3.9 support and cleans up a bit the packaging setup.

TODO:

* [x] @jezdez to add `JAZZBAND_RELEASE_KEY` to the repo secrets.

### CODE REVIEW

File: project/tests/test_view_profiling.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import random
 
 from django.test import TestCase
-from mock import Mock
+unitest. import Mock
```
Comment (auvipy) [inline project/tests/test_view_profiling.py:4]: should this from unittest.mock import

File: project/tests/test_view_requests.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import unittest
 
 from django.test import TestCase
-from mock import Mock
+from unitest.mock import Mock
```
Comment (auvipy) [inline project/tests/test_view_requests.py:5]: from unittest.mock import

File: project/tests/util.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from mock import Mock
+from unitest.mock import Mock
```
Comment (auvipy) [inline project/tests/util.py:1]: from unittest.mock import

File: project/tests/test_view_profiling.py (hunk 1 above)
Comment (jezdez) [reply project/tests/test_view_profiling.py:4]: Opps, yeah, something went wrong there.

**Review by auvipy — APPROVED** [review]
looks good to me

**Review by nasirhjafri — APPROVED** [review]
LGTM 👍 
I just have one concern, last time I checked Github Actions didn't support PRs from forked repos. Do they support now(if yes do we need to enable some flag too for that)?

**Comment by jezdez** [comment]
> LGTM 👍 
> 
> I just have one concern, last time I checked Github Actions didn't support PRs from forked repos. Do they support now(if yes do we need to enable some flag too for that)?

That should be good now, from https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/:

> In order to protect public repositories for malicious users we run all pull request workflows raised from repository forks with a read-only token and no access to secrets.

But to be on the safe side, I'll try later in a different repo to make sure.

**Comment by jezdez** [comment]
> > LGTM 👍
> > I just have one concern, last time I checked Github Actions didn't support PRs from forked repos. Do they support now(if yes do we need to enable some flag too for that)?
> 
> That should be good now, from https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/:
> 
> > In order to protect public repositories for malicious users we run all pull request workflows raised from repository forks with a read-only token and no access to secrets.
> 
> But to be on the safe side, I'll try later in a different repo to make sure.

Ah, no need to do an extra PR, here's a PR that was created from a branch on repo fork: https://github.com/jazzband/django-robots/pull/107

**Pull request merged**